RiskSdk/src/main/java/com/airwallex/risk/AirwallexRisk.kt

```diff
@@ -0,0 +1,149 @@
+            }
+
+            val sharedPreferences = applicationContext.getSharedPreferences(
+                sharedPreferencesFilename, Context.MODE_PRIVATE
+            )
+
+            val storageManager = StorageManager(sharedPreferences = sharedPreferences)
+
+            val riskContext = RiskContext(
+                applicationContext = applicationContext,
+                environment = configuration.environment,
+                tenant = configuration.tenant,
+                storageManager = storageManager
+            )
+
+            val eventRepository = EventRepository()
+
+            val shared = AirwallexRisk(
+                riskContext = riskContext,
+                eventManager = EventManager(
+                    repository = eventRepository,
+                    apiService = ApiService(
+                        riskContext = riskContext
+                    ),
+                    automaticEventProvider = AutomaticEventProvider(
+                        repository = eventRepository,
+                        riskContext = riskContext
+                    )
+                )
+            )
+            this.shared = shared
+            shared.setAccountId(accountId = accountId)
+            shared.start()
+        }
+
+        /**
+         * Sets the signed in Airwallex user ID.
+         * Use this method after user sign in/out to store the user ID to be sent with events.
+         * @param userId Signed in Airwallex user ID. Set `null` on sign out.
+         */
+        fun setUserId(userId: String?) {
+            val shared = shared
+            if (shared == null) {
+                Log.d(Constants.logTag, Constants.logSdkNotStartedError)
+                return
+            }
+            shared.riskContext.updateUserId(userId = userId)
+        }
+
+        /**
+         * Sets the Airwallex account ID.
+         * Use this method to update the account ID if it changes.
+         * @param accountId Airwallex account ID.
+         */
+        fun setAccountId(accountId: String?) {
+            val shared = shared
+            if (shared == null) {
+                Log.d(Constants.logTag, Constants.logSdkNotStartedError)
+                return
+            }
+            shared.riskContext.updateAccountId(accountId = accountId)
+        }
+
+        /**
+         * Adds a new event to the queue.
+         * This is a public method for client apps to log specific lifecycle events, eg. login, logout.
+         * @param event App event that triggered this method call.
+         * @param screen Current app view. Optional.
+         */
+        fun log(event: String, screen: String?) {
+            val shared = shared
+            if (shared == null) {
+                Log.d(Constants.logTag, Constants.logSdkNotStartedError)
+                return
+            }
+            shared.log(eventType = event, screen = screen)
+        }
+    }
+
+    fun start() {
+        Log.d(Constants.logTag, riskContext.description())
```

Comment:
should we remove this for release?

Reply:
It'll only run in debug mode anyway (the `d` is for debug)